content/blog/eng/protected-friday.md

```diff
@@ -0,0 +1,124 @@
+---
+authors: ["engineering"]
+comments: true
+date: "2023-04-17"
+draft: false
+share: true
+categories: [Venerdì Protetto]
+title: "Venerdì Protetto"
+languageCode: "en-US"
+type: "post"
+twitterImage: '/images/logo_engineering.png'
+toc: false
+---
+
+<div id="user-content-toc">
+  <ul>
+    <summary><h1 style="display: inline-block;">Venerdì Protetto</h1></summary>
+  </ul>
+</div>
+
+
+## What is it all about?
+
+At Facile.it we strongly believe in sharing knowledge and experiences. Thus, we dedicate one Friday a month to expanding our knowledge and learning among colleagues in the IT department. During these protected Fridays, better known as Venerdì Protetto, we welcome speakers and participate in talks on different IT-related topics. Speakers who bring to life Venerdì Protetto are colleagues or external guests.
+
+    
+    
+## Curious to learn more?
+
+Here you can find the talks of each Venerdì Protetto.
+
+<table>
+    <thead>
+        <tr
+            align="left"
+            >
+            <th>Date</th>
+            <th>Topic</th>
+            <th>Speakers</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan=4>14/4/2023</td>
+        </tr>
+        <tr>
+            <td><a href="#code-revolution">Code revolution</a></td>
+            <td>Mirko Urru & Stefano Giurgiano</td>
+        </tr>
+        <tr>
+            <td><a href="#dynamic-forms">Dynamic forms</a></td>
+            <td>Giovanni Fiorentino</td>
+        </tr>
+        <tr>
+            <td rowspan=2><a href="#backstage">Backstage</a></td>
+            <td>Lee Mills</td>
+        </tr>
+    </tbody>
+</table>
+
+<br>
+
+## Venerdì Protetto talks
+
+### Dynamic forms
+
+#### Using React with in-house libraries
+
+<sup>by [Giovanni Fiorentino](https://www.linkedin.com/in/giovanni-fiorentino-25004b74/)</sup>
+
+Creating a simple yet flexible and powerful form structure is always a topic of interest in front-end development, especially during an intensive rewriting and refactoring of the codebase.
+
+In this talk, we learned about a need that emerged during the migration to React.js, carried out by our Gas & Power Team. In that circumstance, the team needed to create a set of forms for the Facile.it internal checkout. The forms had to have several similarities but also some differences, depending on the company and the offer. Rather than having many similar components or one that was too complex, our engineers chose to describe the forms through a unique configuration. They consequently built a component capable of designing the modules according to that unique configuration. For this purpose, they used two libraries of the Facile.it design system: Full Metal UI (FMUI) for components and Bilden for validation. This approach to creating forms was successfully applied to the internal checkout and the code was imported into a common library.

```

Comment:
FMUI for layout and design of the form via custom components (non si capisce cosa faccia)